test/core_tests.js

```diff
@@ -1,138 +1,156 @@
+      assert.strictEqual( actual, expected, "canvas.style.transitionDuration initialized correctly" );
+
+      actual = canvas.style.transitionProperty;
+      expected = "all";
+      assert.strictEqual( actual, expected, "canvas.style.transitionProperty initialized correctly" );
+
+      actual = canvas.style.transitionTimingFunction;
+      expected = "ease-in-out";
+      assert.strictEqual( actual, expected, "canvas.style.transitionTimingFunction initialized correctly" );
+
+      actual = document.documentElement.style.height;
+      expected = "100%";
+      assert.strictEqual( actual, expected, "documentElement.style.height is 100%" );
+    });
+
+    QUnit.test( "Steps", function( assert ) {
+      assert.expect( 2 );
+
+      var actual, expected;
+      var step1 = document.querySelector( "div#step-1" );
+
+      actual = step1.style.position;
+      expected = "absolute";
+      assert.strictEqual( actual, expected, "Step position is 'absolute'" );
+
+      assert.ok( step1.classList.contains( "active" ), "Step 1 has active css class." );
+    });
+
+  QUnit.module( "Core API" );
+
+    QUnit.test( ".next()", function( assert ) {
+      assert.expect( 2 );
+
+      var root = document.querySelector( "div#impress" );
+      var step1 = document.querySelector( "div#step-1" );
+      var step2 = document.querySelector( "div#step-2" );
+      var done = assert.async();
+
+      impress().next();
+
+      assert.ok( step2.classList.contains( "active" ), step2.id + " add active css class." );
+      assert.notOk( step1.classList.contains( "active" ), step1.id + " remove active css class." );
+
+      // Reset to original state
+      var removeStepEnterEvent = registerEventListener( root, "impress:stepenter",  function() {
+        removeStepEnterEvent();
+        done();
+      });
+      impress().goto( step1 );
+    });
+
+    QUnit.test( "impress:stepenter event", function( assert ) {
+      assert.expect( 4 );
+
+      var actual, expected;
+      var root = document.querySelector( "div#impress" );
+      var step1 = document.querySelector( "div#step-1" );
+      var step2 = document.querySelector( "div#step-2" );
+      var done = assert.async();
+
+      var removeTestEvent = registerEventListener( root, "impress:stepenter",  function( event ) {
+        actual = event.target;
+        expected = step2;
+        assert.strictEqual( actual, expected, "Triggered event for the second step" );
+
+        assert.ok( step2.classList.contains( "present" ), event.target.id + " add present css class" );
+        assert.notOk( step2.classList.contains( "future" ), event.target.id + " remove future css class" );
+        assert.notOk( step2.classList.contains( "past" ), event.target.id + " remove active css class." );
+
+        removeTestEvent();
+
+        // Reset to original state
+        var removeCleanupEvent = registerEventListener( root, "impress:stepenter",  function() {
+          removeCleanupEvent();
+          done();
```

Comment:
The callbacks are being cascades for explicitness, registering the offending event in the event-loop before each synchronous call. This is definitely a pattern that should be changed in the near future when we have a better way to cleanup `impress.js`.